frontend/src/metabase/lib/entity-id/hooks/use-valid-id.ts

```diff
@@ -0,0 +1,35 @@
+import { P, match } from "ts-pattern";
+
+import {
+  type EntityType,
+  useTranslateEntityIdQuery,
+} from "metabase/api/entity-id";
+
+type ValidIdForEntityProps = {
+  type: EntityType;
+  id: unknown;
+};
+
+export const useValidIdForEntity = ({ type, id }: ValidIdForEntityProps) => {
+  const {
+    data: entity_ids,
+    isError,
+    isLoading,
+  } = useTranslateEntityIdQuery({
```

Comment:
I think we could be a little bit smarter before calling this endpoint which might save time and improve performance. It should be fairly simple to detect if a given ID is an entity ID, it should be comprised of these allowed strings

https://github.com/metabase/metabase/blob/ebf18f35b22d16bc286b3cda9c29b571a815b714/src/metabase/util/jvm.clj#L28

Reply:
+1, if we can have a basic heuristic that detect if it's a valid NanoID first, it could reduce the overhead of this API call.

Reply:
Yep, the only issue is with the stupid rule of hooks in terms of conditionally calling the API. I _think_ I can do something within the API file itself so let me try that first

Reply:
ok, looks like there's an escape hatch with the skip flag so I'll use that